platforms/gke/base/use-cases/inference-ref-arch/kubernetes-manifests/online-inference-gpu-h100-llama4-scout/online-inference-gpu-deployment.yaml

```diff
@@ -0,0 +1,213 @@
+# Copyright 2025 Google LLC
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+# https://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: vllm-h100
+spec:
+  selector:
+    app: vllm-h100
+  type: ClusterIP
+  ports:
+    - protocol: TCP
+      port: 8000
+      targetPort: 8000
+---
+apiVersion: v1
+kind: PodTemplate
+metadata:
+  name: vllm-h100
+template:
+  spec:
+    nodeSelector:
+      cloud.google.com/gke-nodepool: gpu-h100x8-a3h8-dws
```

Comment:
For a cost optimization point of view you should run each model in the machine shape that it demands

This will depend on the model you are running. 
Eg. 
- Gemma3 27B you need only 1 GPUs H100 
- Llama 3 70B you need 4 GPUs H100 
- Llama 4 Scout you need 8 GPUs H100

Reply:
I've seen you have a specific folder per model. That is fine. But we may be able to have a single yaml if we make these things configurable

Reply:
Moved to patch.